### ISSUE TITLE
Fallback to GET for 500 responses.

### ISSUE THREAD

**Issue by Pigankle** [DjangoAdminHackers/django-linkcheck#224]
https://github.com/DjangoAdminHackers/django-linkcheck/blob/c13ea894c153d5c18314c4d030f3fb52d499563a/linkcheck/models.py#L413

Some servers return 500 to HEAD requests but 200 to GET. The current fallback only retries with GET on 400 and 405. This adds 500 to that list so those URLs will be rechecked (and not incorrectly marked as broken.)
An example of such a site:
https://swefc.unm.edu/home/resources/?_sft_topic=water-conservation

```sh
# Fails (HEAD request - what linkcheck uses)
curl -I "https://swefc.unm.edu/home/resources/?_sft_topic=water-conservation"

# Succeeds (GET request - what browsers use)
curl -L "https://swefc.unm.edu/home/resources/?_sft_topic=water-conservation"
```

I think the fix would be changing line 413 from 

```py
            if response.status_code in [HTTPStatus.BAD_REQUEST, HTTPStatus.METHOD_NOT_ALLOWED]:
```
to
```py
            if response.status_code in [HTTPStatus.BAD_REQUEST, HTTPStatus.METHOD_NOT_ALLOWED, HTTPStatus.INTERNAL_SERVER_ERROR]:
```

**Comment by Pigankle** [comment]
I have a fork with that single change in it - if you like it I can PR it.

**Comment by claudep** [comment]
Please do!

**Comment by Pigankle** [comment]
https://github.com/DjangoAdminHackers/django-linkcheck/pull/225
